source/materials/OpticalMaterialProperties.cc

```diff
@@ -736,6 +736,105 @@ namespace opticalprops {
   }
 
 
+  /// PTP (p-terphenyl) ///
+  G4MaterialPropertiesTable* PTP()
+  {
+    // Data from https://doi.org/10.1016/j.nima.2011.12.036
+    // and https://iopscience.iop.org/article/10.1088/1748-0221/5/04/P04007/
+    G4MaterialPropertiesTable* mpt = new G4MaterialPropertiesTable();
+
+    // REFRACTIVE INDEX
+    std::vector<G4double> rIndex_energies = {optPhotMinE_, optPhotMaxE_};
+    std::vector<G4double> PTP_rIndex      = {1.65    , 1.65};
+    mpt->AddProperty("RINDEX", rIndex_energies, PTP_rIndex);
+
+    // ABSORPTION LENGTH
+    // Assuming no absorption except WLS
+    std::vector<G4double> abs_energy = {optPhotMinE_, optPhotMaxE_};
+    std::vector<G4double> absLength  = {noAbsLength_, noAbsLength_};
+    mpt->AddProperty("ABSLENGTH", abs_energy, absLength);
+
+
+
+    // WLS ABSORPTION LENGTH
+    // There are no tabulated values in the literature for the PTP absorption
+    // length as a function of the wavelength. 
+    // However in https://iopscience.iop.org/article/10.1088/1748-0221/5/04/P04007/
+    // it says that "the thickness >=150 nm of a polycrystalline p-terphenyl layer 
+    // is enough for absorption of >=99.9% of the xenon light". Thus, using the
+    // formula P = e^(-x/lambda) where lambda is the absorption length, we can
```

Comment:
To be clearer, I would say: `1-P = e^(-x/lambda), where P is the absorption probability`.